application/classes/Ushahidi/Rest.php

```diff
@@ -266,7 +266,9 @@ protected function _check_access()
 		}
 		catch (OAuth2Exception $e)
 		{
-			if (!$this->_is_auth_required() AND $e instanceof MissingAccessTokenException)
+			if (!$this->_is_auth_required() AND (
+				$e instanceof MissingAccessTokenException OR
+				$e instanceof InvalidAccessTokenException))
```

Comment:
@tuxpiper I've reverted this. But I'm not clear how important that change was for caching?